lib/heuristics/concerns/scheduling_data_initialisation.rb

```diff
@@ -305,4 +303,18 @@ def compute_capacities(quantities, vehicle, available_units = [])
 
     capacities
   end
+
+  def common_divisor?(lapses)
+    # If lapses have common multiplexer then any of these
```

Comment:
I think `common_divisor?` naming is more appropriate for a function that checks if the caller has a common divisor -- like `5.common_divisor?(3)` 
```suggestion
  def has_common_divisor?(lapses)
    # If lapses have a common divisor then any of these
```